__tests__/calcIntegration.test.ts

```diff
@@ -1,11 +1,11 @@
 import { promises as fs } from 'fs'
 import { parseStringPromise } from 'xml2js'
-import { convertToXML } from '../src/lib/xml/xmlConverter'
-import { calculate } from '../src/lib/calculation'
-import { TaxFormUserInput } from '../src/types/TaxFormUserInput'
-import { PostponeUserInput } from '../src/types/PostponeUserInput'
-import { convertPostponeToXML } from '../src/lib/postpone/postponeConverter'
-import { setDate } from '../src/lib/utils'
+import { convertToXML } from '../src/features/summary/9-pokracovat/xml/xmlConverter'
```

Comment:
Zmena struktury mi pride ako osobna preferencia. Mne pride povodna struktura prehladnejsia. Ale ak sa na tom zhodnete s ostatnymi developermi tak s tym nemam problem. Len by som bol rad ak sa na takejto zmene zhodnete viaceri ze vam to takto vyhovuje.

Reply:
co tym myslis ze osobna preferencia? je to standard svetovy robit feature foldre aka micro-servisy, namiesto delenia veci podla typu a potom vsetko jedneho typu je na jednom mieste, toto je proste ze mas akoby kazdu samostatne fungujucu vec ako vlastny mini react projekt, a vies ze vsetko je tam a nie je problem si celu jednu pagu nastudovat lebo nie su ziadne externe veci ktorym by si musel rozumiet - trvalo mi niekolko hodin pochopit to, zatial co ked to je takto, tak za 5 minut pochopis jeden screen a za dalsich 5 dalsi screen, a neni tam onboarding bariera

ale nerozumiem co myslis osobna preferencia, z coho sudis ze to je svojska architektura? pozeral si sa na tu strukturu a zamyslal si sa nad plusmi a minusmi? pripomina ti to nejaky standard ktory sa uz niekde pouziva? co si o tom standarde myslis? ake su jeho plusy a minusy?

pride mi zvlastne ze na standard povies ze osobna preferencia, nie pretoze by som teraz chcel ist toto davat ako standard ktory je lepsi, ale preto ze ked nieco nepoznas, tak podla mna krok jedna by bol spytat sa ze co je zamer, a az krok dva to sudit - lebo vsetko co nepoznas vyzera na prvy pohlad ako osobna preferencia, ale ta osobna preferencia ma dovody

dobre rozumiem ze hovoris ze ta osobna preferencia je arbitrarna a neprinasa ziadne vyhody? to si myslis alebo to je vyzva na diskusiu ze kde bol tento system standard vyskusany a preverovany a kolko rokov a na kolkych developeroch, ze si ho niekto osobne preferuje?

ano, viaceri by v tom mali vidiet zmysel, resp ti ktori na tom budu robit a prevezmu zodpovednost - ja som to povodne dal ako navrh julesovi, a ked videl ukazku tak povedal ze tam vidi nejake plusy, a tak som to tak spravil cele, teraz viem ze hocikto pride, tak nemusi pozerat cely kod, staci pozriet jeden screen a vsetko suvisiace tam vidi, je v jednom foldri, ostatne nemusi poznat - dovody preco toto preferovat ti viem rozpravat z praxe alebo ti dam kontakt na desiatky ludi ktori kazdy ti povie svoju skusenost pred a po feature-foldroch

Reply:
> je to standard svetovy robit feature foldre aka micro-servisy

toto nie je aplikacia postavena na micoservice architekture

samotny framework next.js ma nejake zakladne defaults kde chce mat ktore subory a kazdi nextjs developer ked sa pozrie na aktualny folder structure tak vie co kde je

tento projekt ma vacsie problemy ako folder strutuce a uz mame niekolko developerov ktori si nastudovali aktualnu folder strutuce a su pripraveni nad nou robit a teraz by si to museli cele prechadzat znova

Reply:
> ale nerozumiem co myslis osobna preferencia, z coho sudis ze to je svojska architektura?
Prepac, nemyslel som svojska architektura. Tiez mame v praci velke projekty ktore maju podobnu architekturu ako tato co navrhujes a dava to tam zmysel. Myslel som to cisto ako moj nazor na tieto zmeny.

> dobre rozumiem ze hovoris ze ta osobna preferencia je arbitrarna a neprinasa ziadne vyhody?
Nie, tak som to nemyslel. Podla mna je projekt dostatocne maly na to aby mu stacila jednoduchsia (aktualna) atchitektura. Ale mozno sa mylim, lebo som na projekte co-to robil a poznam ho a tak som tym ovplyvneny.

Ak si projekt uz nastudovalo viacero developerov tak si nemyslim ze zmena struktury je teraz na mieste - teraz ked treba co najrychlejsie spravit verziu na novy rok. Takyto refactoring podla mna dava zmysel neskor, po termine danoveho priznania napr. v 2. polovici roka, ked sa uz nebude robit na funkcionalite. Neviem aka bola diskusia o tomto refactoringu, preto som napisal ze by som bol rad ak by sa na tom zhodol cely tim (resp vacsina alebo team leadi). Ak je to tak, potom absolutne nemam ziadne vyhrady. Ja nemam "kluce od miesacky" takze ak sa na tom zhodnete, moj nazor moze byt irelevantny a nebude mi to vadit.

> ja som to povodne dal ako navrh julesovi
Ako som napisal - mne je to prakticky jedno, obe architektury so dobre. Ak sa tak dohodnete tak to kludne refactorujte. Ja na tomto projekte nebudem robit "full time", iba vam obcas do toho budem kecat v pull requestoch prip pomozem s nejakymi drobnostami ak sa nebudete nahodou vediet pohnut 😃

Reply:
neviem marek na co z toho co pises chces aby som zareagoval, suhlasim ze nie je microservisova architektura, a ani neni ciel robit taku, ciel je mat separation of concern

ci su tam vacsie problemy je taka zvlastna perspektiva, vacsia urcite su, ale onboarding novych ludi je podla mna vecny problem vzdy a vsade, a robit kod feature oriented bude len plusom, ci uz maintanence alebo onboarding, minus tam nevidim

neviem ako poznanie next rozdelenia po typoch pomoze programatorovi zorientovat sa vo featurach projektu a nezavislsoti kodu a teda znizeneho rizika pri upravovani kodu, ze vies ze nezasiahnes do nicoho ineho, lebo vzdy je maly celok pokope ktory nesuvisi so zvyskom

myslim ze ta struktura je lepsia aj pre tych co uz s tym robili, jules s tym bol ok, mam to vyskusane, a mam ambiciu zapojit juniorov, a viem ze toto odstranuje bariery, nevidim ziadne minus a jediny argument vidim status quo

Reply:
@mathio som rad ze poznas podobnu architekturu, a ze rozumies jej vyhodam - mohol by si to prosim zhrnut aj pre mareka? lebo myslim ze mas k jeho vnimaniu blizsie

ta architektura je super aj na malych projektoch, pokial nerobis one pager, ale nieco dlhodobe, tak je to len a len plus, jedina "nevyhoda" je ze to tak musis nastavit, ale to je zlomok casu co potom stravis na projekte, a potom prichadzaju uz len plusy

nemyslim ze tym co poznaju kod by "lepsia" architektura mala spravit prekazku len preto ze si zvykli, a optimalizoval by som nie na starych ale na zapajanie a rozdelovanie prace

suhlasim s tym terminom ze nerobit to natesno, preto som to pod julesovym dohladom robil velmi rychlo a intenzivne a vsetok feedback do dna zapracovaval - kazdopadne ale sa netreba bat, pretoze nic sa tam nemenilo, iba organizacia suborov je taka, ze hned z nej pochopis co vsetko sa v projekte nachadza, predtym bolo vsetko nakope, a to nie je dobre, najma ked ten projekt uz je dost velky, nie nejaka 3 dnova robota

diskusia nebola pretoze aj marek aj matej ste napisali ze nemate cas, jediny jules mal cas, a daily som to s nim riesil, az kym povedal ze je to ready na merge, ale chcel este nazor

suhlasim ze takuto zmenu by som nerobil na poslednu chvilu zbytocne, na druhej strane ak chces rozdelit pracu medzi ludi, potrebujes to mat feature-oriented, a je to done, neni to ze sa vsetko rozbilo, testy som priebezne pustal, a zmeny som robil len organizacne, nie implementacne

(co sa tyka tych juniorov co povedali ze kodu rozumeju, tak ja tam mam otazky aj po hodinach diskusii s julesom, ktory to vytvoril, a po refaktorovani kde som musel vsetko pochopit a riesil som to kazdy den intenzivne, takze tomu neverim ani nahodou ze by sa vyznali, ale aj keby, tu sa budu vyznat lepsie, a aj keby rovnako, tak tu sa bude lahko rozdelovat praca a lahsie onboardovat, takze opat som nazoru ze minusy to nema - jedine doladit tie ostatne feedbacky, to urcite treba nech to neni ze veci budu horsie, ale nech je vsetko min tak dobre ako bolo, kde nie je tak to bud vratme alebo zlepsime)